samples/SharpDx2D/MainWindow.xaml.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+using SharpDX.Direct2D1;
+
+namespace SharpDx2D
+{
+    /// <summary>
+    /// Interaction logic for MainWindow.xaml
+    /// </summary>
+    public partial class MainWindow : Window
+    {
+        private TimeSpan lastRenderTime;
+        private RenderTarget renderTarget = null;
+        private SharpDX.DirectWrite.TextLayout textLayout;
+        private SharpDX.Direct2D1.Brush brush;
+
+        public MainWindow()
+        {
+            InitializeComponent();
+            Loaded += MainWindow_Loaded;
+            SizeChanged += MainWindow_SizeChanged;
+        }
+
+        private void MainWindow_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            Grid g = (Grid)Content;
+            if (g.ActualWidth > 0 && g.ActualHeight > 0)
+            {
+                InteropImage.SetPixelSize((int)g.ActualWidth, (int)g.ActualHeight);
+            }
+            else
+            {
+                Dispatcher.BeginInvoke(System.Windows.Threading.DispatcherPriority.Background, (Action<object, SizeChangedEventArgs>)MainWindow_SizeChanged, null, null);
+            }
+        }
+
+        private void MainWindow_Loaded(object sender, RoutedEventArgs e)
+        {
+            SharpDX.DirectWrite.Factory factory = new SharpDX.DirectWrite.Factory(SharpDX.DirectWrite.FactoryType.Isolated);
+            SharpDX.DirectWrite.TextFormat textFormat = new SharpDX.DirectWrite.TextFormat(factory, "Arial", 30);
+            textLayout = new SharpDX.DirectWrite.TextLayout(factory, "Test", textFormat, 0, 0);
+            MainWindow_SizeChanged(null, null);
+            InteropImage.WindowOwner = (new System.Windows.Interop.WindowInteropHelper(this)).Handle;
+            InteropImage.OnRender = OnRender;
+            CompositionTarget.Rendering += CompositionTarget_Rendering;
+        }
+
+        private void OnRender(IntPtr handle, bool isNewSurface)
+        {
+            if (isNewSurface)
+            {
+                if (brush != null)
+                {
+                    brush.Dispose();
+                    brush = null;
+                }
```

Comment:
style nit. can we leave a blank line after a closing brace, if the next line isn't another closing brace?